lib/RestClient.ts

```diff
@@ -19,7 +19,9 @@ export interface IRequestOptions {
     // since accept is defaulted, set additional headers if needed
     additionalHeaders?: ifm.IHeaders,
 
-    responseProcessor?: Function
+    responseProcessor?: Function,
+    //Dates aren't automatically deserialized by JSON, this adds a date reviver to ensure they aren't just left as strings
+    reviveDates?: boolean
```

Comment:
Is revive a common name for this process? Deserialize might be more helpful to consumers to know what it's doing?

Reply:
Or maybe there's something else. Revive is a bit weird, may be just me.

Reply:
Most of the documentation I've seen refers to it as a reviver (e.g. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse). Not sure why, maybe people were encoding/decoding dates in different formats and they wanted to make clear that this wasn't a standard part of their deserialization?

Nonetheless, I agree deserialize is more intuitive to me and maybe more accurate in this case since we aren't letting them provide their own custom date encoding/decoding. I'll switch it over.